vochain/process.go

```diff
@@ -143,8 +146,30 @@ func (v *State) CountProcesses(committed bool) (uint64, error) {
 	return count, nil
 }
 
+// ListProcessIDs returns the full list of process identifiers (pid).
+func (v *State) ListProcessIDs(committed bool) ([][]byte, error) {
+	if !committed {
+		v.Tx.RLock()
+		defer v.Tx.RUnlock()
+	}
```

Comment:
instead of taking a boolean parameter to tell the method whether or not to grab the lock, what I would do is make it the caller's responsibility to hold the lock. this way it is more explicit, as you don't have to remember what the boolean does.

Reply:
SGTM